app/controllers/user_leagues_controller.rb

```diff
@@ -0,0 +1,22 @@
+class UserLeaguesController < ApplicationController
+  before_action :authenticate_user!
+  validates :user_id, :league_id, presence: true
+  validates :user_id, uniqueness: { scope: :league_id }
+  belongs_to :user
```

Comment:
Les belongs to sont déjà dans le model de user-league il ne faut pas les rajouter dans le controller.